hbase-server/src/main/java/org/apache/hadoop/hbase/wal/BaseRecoveredEditsOutputSink.java

```diff
@@ -0,0 +1,281 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.hadoop.hbase.wal;
+
+import static org.apache.hadoop.hbase.wal.WALSplitUtil.getCompletedRecoveredEditsFilePath;
+import static org.apache.hadoop.hbase.wal.WALSplitUtil.getRegionSplitEditsPath;
+
+import java.io.EOFException;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+import java.util.concurrent.ConcurrentHashMap;
+
+import org.apache.hadoop.conf.Configuration;
+import org.apache.hadoop.fs.FileSystem;
+import org.apache.hadoop.fs.Path;
+import org.apache.hadoop.hbase.Cell;
+import org.apache.hadoop.hbase.CellUtil;
+import org.apache.hadoop.hbase.HConstants;
+import org.apache.hadoop.hbase.log.HBaseMarkers;
+import org.apache.hadoop.hbase.util.Bytes;
+import org.apache.hadoop.hbase.util.FSUtils;
+import org.apache.hadoop.ipc.RemoteException;
+import org.apache.yetus.audience.InterfaceAudience;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import org.apache.hbase.thirdparty.org.apache.commons.collections4.MapUtils;
+
+@InterfaceAudience.Private
+public abstract class BaseRecoveredEditsOutputSink extends OutputSink {
+  private static final Logger LOG = LoggerFactory.getLogger(RecoveredEditsOutputSink.class);
+  private WALSplitter walSplitter;
+  private FileSystem walFS;
+  private Configuration conf;
+  private final ConcurrentHashMap<String, Long> regionMaximumEditLogSeqNum =
+      new ConcurrentHashMap<>();
+
+  public BaseRecoveredEditsOutputSink(WALSplitter walSplitter,
+      WALSplitter.PipelineController controller, EntryBuffers entryBuffers, int numWriters) {
+    super(controller, entryBuffers, numWriters);
+    this.walSplitter = walSplitter;
+    this.walFS = walSplitter.walFS;
+    this.conf = walSplitter.conf;
+  }
+
+  /**
+   * @return a path with a write for that path. caller should close.
+   */
+  protected RecoveredEditsWriter createWriter(byte[] region, WAL.Entry entry) throws IOException {
```

Comment:
Odd passing an 'entry' when we are creating recepticle for entries? Pass the path extracted from Entry and sequenceid instead?

Reply:
Will change to pass the tablename, regionname, and the sequenceid.